framework/include/internal/fwk_core.h

```diff
@@ -32,6 +32,11 @@ int __fwk_init(size_t event_count);
  */
 noreturn void __fwk_run(void);
 
+/*
+ * \brief Processing events already raised by modules and interrupt handlers.
+ */
+void __fwk_run_event(void);
```

Comment:
Please rename this to `void fwk_process_event_queue(void)` to make it clear.
Also, this can be the public API used externally by the framework and other modules.